registry/modules/auth-basic/utils/jwt/tokenManager.ts

```diff
@@ -0,0 +1,60 @@
+import jwt from "jsonwebtoken";
+
+import { jwtEnvVariablesMissing } from "@/modules/auth-basic/utils/errors/auth.js";
+
+/**
+ * Example reading public/private keys from a "./keys" folder in rootDir
+ *
+ * const JWT_PUBLIC_KEY = fs.readFileSync(path.join(process.cwd(), "/keys/jwt_public.key"), { encoding: 'utf8' });
+ * const JWT_PRIVATE_KEY = fs.readFileSync(path.join(process.cwd(), "/keys/jwt_private.key"), { encoding: 'utf8' });
+ */
+const JWT_ISSUER = process.env.JWT_ISSUER; // This is set as the project name in `package.json`
+const JWT_SECRET_KEY = process.env.JWT_SECRET_KEY as string; // Randomly generated when the API module is installed. Min length: 64 characters
+
+/**
+ * JWT Token Manager - Define any JWT-related functions here to reuse JWT configuration
+ *
+ * @see {algorithm} Use the `RS256` (or equivalent) algorithm with public/private keys if you are building a distributed system or the JWT token will be shared between services
+ */
+const TokenManager = <T extends string | Buffer | object>(
+  secretOrPrivateKey: string,
+  secretOrPublicKey: string,
+  options: jwt.SignOptions | jwt.VerifyOptions
+) => {
+
+  // TODO: Replace with a package that checks if all .env variables in .env.sample are configured on build
+  if (!JWT_SECRET_KEY || !JWT_ISSUER) {
+    throw jwtEnvVariablesMissing();
```

Comment:
Idealy we have some logic that checks if all environment variables are present when the service starts. Currently this exception would only be thrown when the TokenManager class is called I believe.